lib/devices/gadgets/fitness_equipment.dart

```diff
@@ -198,7 +198,8 @@ class FitnessEquipment extends DeviceBase with PowerSpeedMixin {
     }
 
     merged.timeStamp = DateTime.now();
-    return merged;
+    final clone = RecordWithSport.clone(merged);
```

Comment:
Do you remember why the clone was needed?

Reply:
If a piece of equipment has low resolution counters (like my Schwinn), then duplicate packets can be received because the equipment sends packets more frequently than the counters are updated.  This leads to the possibility that `merged` is the same object as `lastRecord` after the assignment at line 190.  

Then `merged` (which is the same as `lastRecord`) is passed as the `stub` parameter to `processRecord()`.  Then the assignment of `elapsedMillis` to `stub.elapsedMillis` is actually setting `lastRecord.elapsedMillis`.  Then at line 990, the `dTMillis` calculation is 0 and which means that at line 1002 `stub.movingTime` does not change.  For me, this results in significant lost moving time.

By cloning `merged`, it is guaranteed that `stub` and `lastRecord` are different objects with different timestamps and moving time won't be lost.

Reply:
Thanks for the explanation. I'll revise it, the worst case it might cause negligible performance penalty. But regardless I kept it in my PR to your branches. So merge those and then we can merge yours.